runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/config/BindingConfigBuilder.java

```diff
@@ -116,6 +117,18 @@ public BindingConfigBuilder<T> options(
         return this;
     }
 
+    public CatalogRefConfigBuilder<BindingConfigBuilder<T>> catalog()
+    {
+        return new CatalogRefConfigBuilder<>(this::catalog);
+    }
+
+    public BindingConfigBuilder<T> catalog(
+        CatalogRefConfig catalogRef)
```

Comment:
Do we need `CatalogRefConfig`?
Perhaps we can use `List<CatalogedConfig>` directly instead to keep it simpler?